### DISCUSSION TITLE
RabbitMQ 3.13.3 ErLang 26.2.5 - Management UI not showing data

### DISCUSSION THREAD

**Discussion by jforward5** [rabbitmq/rabbitmq-server#11502] [Questions]
Hello RabbitMQ peeps!

I just upgraded my RabbitMQ instance from 3.12 to 3.13.3 on Monday.  After the upgrade my Management UI was working until I accessed the API by simply visiting the url.  I visited the endpoint that shows the connection information (/api/connections).

After hunting around the web to try and find out what would cause this behavior, I found the suggestions here for clearing the browser cache, so I did that.  It didn't help.  Next I went through the settings for the instance to ensure that it had the management plugin enabled, and that my user has access to all information, and just to make sure I granted global permissions and set the user as administrator.

Since none of those things worked, I then rebuilt the docker container that is hosting RMQ, no love there either.

Next I looked at the browser console to see if there were errors, and there is an error:
![image](https://github.com/rabbitmq/rabbitmq-server/assets/49663221/30fe736d-34fc-4f10-a106-8819272d7b39)

I tried other browsers too to see if it was a FireFox issue, and the same error is present no matter the browser.  This feels like something simple that I have configured that isn't compatible with 3.13.3...

I tried to access the api endpoint referenced and it doesn't seem to exist.  The "api/depricated-features" exists, just not the "/used" portion.
For clarity, I am using the "rabbitmq:management" docker container.

Does anyone know how I can fix this?

**Comment by michaelklishin** [comment]
Both endpoints [do exist](https://github.com/rabbitmq/rabbitmq-server/blob/v3.13.x/deps/rabbitmq_management/src/rabbit_mgmt_dispatcher.erl#L172-L173) and they are not registered conditionally.

This is very easy to verify with a local 3.13.3 node with the management plugin enabled:

```
xh --json http://guest:guest@localhost:15672/api/deprecated-features

HTTP/1.1 200 OK
Cache-Control: no-cache
Content-Encoding: gzip
Content-Length: 309
Content-Security-Policy: script-src 'self' 'unsafe-eval' 'unsafe-inline'; object-src 'self'
Content-Type: application/json
Date: Wed, 19 Jun 2024 20:38:05 GMT
Server: Cowboy
Vary: accept, accept-encoding, origin

[
    {
        "name": "management_metrics_collection",
        "desc": "",
        "deprecation_phase": "permitted_by_default",
        "doc_url": "https://blog.rabbitmq.com/posts/2021/08/4.0-deprecation-announcements/#disable-metrics-delivery-via-the-management-api--ui",
        "provided_by": "rabbitmq_management_agent"
    },
    {
        "name": "classic_queue_mirroring",
        "desc": "",
        "deprecation_phase": "permitted_by_default",
        "doc_url": "https://blog.rabbitmq.com/posts/2021/08/4.0-deprecation-announcements/#removal-of-classic-queue-mirroring",
        "provided_by": "rabbit"
    },
    {
        "name": "global_qos",
        "desc": "",
        "deprecation_phase": "permitted_by_default",
        "doc_url": "https://blog.rabbitmq.com/posts/2021/08/4.0-deprecation-announcements/#removal-of-global-qos",
        "provided_by": "rabbit"
    },
    {
        "name": "ram_node_type",
        "desc": "",
        "deprecation_phase": "permitted_by_default",
        "doc_url": "https://blog.rabbitmq.com/posts/2021/08/4.0-deprecation-announcements/#removal-of-ram-nodes",
        "provided_by": "rabbit"
    },
    {
        "name": "transient_nonexcl_queues",
        "desc": "",
        "deprecation_phase": "permitted_by_default",
        "doc_url": "https://blog.rabbitmq.com/posts/2021/08/4.0-deprecation-announcements/#removal-of-transient-non-exclusive-queues",
        "provided_by": "rabbit"
    }
]
```

```
xh --json http://guest:guest@localhost:15672/api/deprecated-features/used

HTTP/1.1 200 OK
Cache-Control: no-cache
Content-Length: 2
Content-Security-Policy: script-src 'self' 'unsafe-eval' 'unsafe-inline'; object-src 'self'
Content-Type: application/json
Date: Wed, 19 Jun 2024 20:38:10 GMT
Server: Cowboy
Vary: accept, accept-encoding, origin

[]
```

You may be using a proxy that only proxies one of the two endpoints.

**Reply by jforward5** [reply]
Here is my "proxy" in nginx:

server {
		listen          443;
		server_name     xxxx.xxxx.com;
		location ~* /api/(.*?)/(.*) {
			client_body_buffer_size 128k;
			proxy_send_timeout   90;
			proxy_read_timeout   90;
			proxy_buffer_size    4k;
			proxy_buffers     16 32k;
			proxy_busy_buffers_size 64k;
			proxy_temp_file_write_size 64k;
			proxy_connect_timeout 30s;
			proxy_pass      http://localhost:15672/api/$1/%2F/$2?$query_string;
			#rewrite ^/rabbitmq/(.*)$ /$1 break;
			proxy_buffering                    off;
			proxy_set_header Host              $http_host;
			proxy_set_header X-Real-IP         $remote_addr;
			proxy_set_header X-Forwarded-For   $proxy_add_x_forwarded_for;
			proxy_set_header X-Forwarded-Proto $scheme;
		}
		
		location ~* /(.*) {
			rewrite ^/rabbitmq/(.*)$ /$1 break;
			proxy_pass http://localhost:15672;
		}
}

From your example both those endpoints are coming from the same "api" so it seems to reason that my browser should be able to find the "usage" endpoint, since it is in the same api, unless my proxy configuration is incorrect...

**Comment by michaelklishin** [comment]
There were no [relevant changes in `3.13.3`](https://github.com/rabbitmq/rabbitmq-server/releases/tag/v3.13.3).
